### DISCUSSION TITLE
Add explicit `index.html` for each permalink?

### DISCUSSION THREAD

**Discussion by AcrylicShrimp** [Elderjs/elderjs#223] [Ideas] [answered]
Hello! I'm using elderjs with AWS S3 and Cloudfront. In this infra, I have to append `index.html` for each permalinks(they don't serve files that have trailing slashes). So I want to build a plugin that handles it automatically. Does it sound bad?

Thank you!

**Comment by nickreese** [comment, marked as answer]
This already happens. For a permalink of `/foo/`, a file of `/foo/index.html` is generated.

**Reply by AcrylicShrimp** [reply]
> This already happens. For a permalink of `/foo/`, a file of `/foo/index.html` is generated. 

No, I mean, for link generation. A common way to add links to internal pages is using permalink helper function. But it currently not support to add index.html at the end.

**Reply by nickreese** [reply]
Ah, I get what you are saying. Have you tried s3 bucket hosting? It turns that on as I've used it in the past.

I don't think this would be easy to do via a plugin without regex, the best place to do this is in `wrapPermalink.ts` I'd be open for a PR for it as long as tests cover it.

**Reply by AcrylicShrimp** [reply]
I tried modifying `wrapPermalink.ts`, but it seems to affect many modules(such as creating `index.html`, so on). I will try more and open PR when I think its ready. Thank you! :)
